File: src/main/java/com/example/appforauslenderamt/entity/PreviousStaysInGermany.java
package com.example.appforauslenderamt.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Value;

@Value
@Builder
@Getter
@AllArgsConstructor
public class PreviousStaysInGermany {

    @JsonProperty("from_date")
    String fromDate;
    @JsonProperty("to_date")
    String toDate;
    @JsonProperty("place")
    String place;
    @JsonProperty("distrinct")
    String district;
    @JsonProperty("state")
    String state;
    @JsonProperty("land")
    String land;

    @Override
    public String toString() {
        return place + ", " + district + ", " + state + ", " + land;
    }
}

File: Frontend/src/app/Models/BackendModels/PreviousStaysInGermany.ts
import { Expose } from "class-transformer"

export class PreviousStaysInGermany {
    @Expose()
    from_date: string
    @Expose()
    to_date: string
    @Expose()
    place: string
    @Expose()
    distrinct: string
    @Expose()
    state: string
    @Expose()
    land: string

    constructor(
        fromDate: string,
        toDate: string,
        place: string,
        district: string,
        state: string,
        land: string
    ) {
        this.from_date = fromDate
        this.to_date = toDate
        this.place = place
        this.distrinct = district
        this.state = state
        this.land = land
    }
}
